staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -91,6 +91,12 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 			KeepAlive: 30 * time.Second,
 		}).DialContext
 	}
+
+	// If we use a cert callback, we need to handle certificate rotation properly
+	if !config.HasCertAuth() && config.HasCertCallback() {
```

Comment:
I suspect that this means that if you clear the file names (and I think I saw you do that above https://github.com/kubernetes/kubernetes/pull/79083/files#diff-99943b34a1da648702fb7e8ca8540b21R96-R97), then even if you have transports with different base files, your cache function will be the same and you could return the wrong content.

It's not obvious to me why the filenames need to be cleared.    I would expect the 
conversion to happen when computing the TLSConfig.

Reply:
Also, a test for this exact case would put my mind at ease.  Different filenames going into `func (c *Config) TransportConfig() (*transport.Config, error)` and then the resulting tlsConfigKeys being different.

Reply:
The key is based on the function pointer of GetCert, not the filenames, and we create an anonymous GetCert function for every call to `TransportConfig`, so actually I think its the inverse; the keys will always differ, even if the filenames are the same. I don't know if that is an issue. I'll write a testcase

Reply:
I see what you mean, that may indeed be a problem. I was misunderstanding what getting the function pointer in the cache key would acheive. I was removing the filenames because otherwise you will end up with them being parsed into certdata and keydata in the transport package, and this means the callback isn't used - but I can sort that out instead